maths/rkf45.py

```diff
@@ -0,0 +1,87 @@
+from collections.abc import Callable
+
+import numpy as np
+
+
+class RangeError(Exception):
+    "Will be raised when initial x is greater than or equal to final x"
+
+
+def runge_futta_fehlberg_45(
+    ode: Callable,
+    x_initial: float,
+    y_initial: float,
+    step_size: float,
+    x_final: float,
+) -> np.ndarray:
+    """
+    Solve ODE using Runge-Kutta-Fehlberg Method (rkf45) of order 5.
+
+    Reference: https://en.wikipedia.org/wiki/Runge%E2%80%93Kutta%E2%80%93Fehlberg_method
+
+    args:
+    ode : Ordinary Differential Equation as function of x and y.
+    x_initial : Initial value of x.
+    y_initial : Initial value of y.
+    step_size : Increment value of x.
+    x_final : Final value of x.
+
+    Returns:
+        np.ndarray: Solution of y at each nodal point
+
+    # exact value of y[1] is tan(0.2) = 0.2027100355086
+    >>> def f(x,y):
+    ...     return 1+y**2
+    >>> y=runge_futta_fehlberg_45(f, 0, 0, 0.2, 1)
+    >>> y[1]
+    0.2027100937470787
```

Comment:
We will need more than one test.  What happens if `x_initial` is set to 5, -5, 5.5?

What happens if `step_size` is zero or negative?

What happens if `x_final` is zero or negative?

What happens if `ode` is `5` instead of being a function?

Let's have a test that raises a RangeError.